packages/documentation/src/pages/getting-started/common-tasks/new-end-to-end-test.mdx

```diff
@@ -0,0 +1,153 @@
+      data.veteranFullName.first,
+    )
+    .fill('input[name="root_veteranFullName_last"]', data.veteranFullName.last)
+}
+```
+_These are recommendations not requirements._
+
+- separate navigation from field input
+  - use a **main test file** for navigation, assertions, and calls helpers
+  - use a **helper file** that for filling out forms
+- create separate, numbered **main test files** to organize tests by their focus
+  - **00-all-fields.e2e.spec.js** - required and optional fields
+  - **01-required.e2e.spec.js** - only required fields
+  - **02-accessibility.e2e.spec.js** - validates accessibility
+  - **03-auth.e2e.spec.js** - validates authentication
+  - **04-cross-cutting-feature.e2e.spec.js** - validates one feature used across several pages (e.g. save in progress)
+- export the end to end test using `E2eHelpers.createE2eTest()`- see [Helpers](#helpers)
+- group tests by pages and use a comment to indicate what page is being tested
+- mock all api responses before starting the test. See [Mocking API responses](#mockimg-api-responses)
+- use `waitForElementVisible` before interacting with any element on the page
+- use `Timeouts` constants for setting timeouts (`platform/testing/e2e/timeouts.js`)
+- use helpers for filling data and performing actions on the page
+- perform `axeCheck` on the main body of the application on each page- see [Custom Nightwatch commands](#custom-nightwatch-commands)
+- assert that each navigation is successful
+
+## Mocking API responses
+
+**my-application-helpers.js**
+
+A mock server runs with the end to end tests to allow tests to make production-like calls.
+
+```
+const mock = require('platform/testing/e2e/mock-helpers');
+
+function initApplicationSubmitMock() {
+  mock(null, {
+  // path to mock
+    path: '/v0/my-application',
+  // verb to mock
+    verb: 'post',
+  // mock response
+    value: {
+      formSubmissionId: '123fake-submission-id-567',
+      timestamp: '2016-05-16',
+    },
+  });
+}
+```
+
+- `mock-helpers.js` for mocking data
+  - mock server is started as part of the end to end testing script
+- mock all api calls in your application e.g. submit and upload calls
+
+## Custom Nightwatch commands
+
+Nightwatch supports extending its client api with [custom commands](https://nightwatchjs.org/guide/#writing-custom-commands). Custom commands are located in `src/platform/testing/e2e/nightwatch-commands`
+
+_Command commands are available on the Nightwatch client e.g. `client.axeCheck()`. Below are some of the commonly used custom Nightwatch commands._
+
+- [`axeCheck(selector)`](https://github.com/department-of-veterans-affairs/vets-website/blob/d721cf0581c6b7d6f26903540dfc27f26a16a7be/src/platform/testing/e2e/nightwatch-commands/axeCheck.js#L4-L7) - uses the [`axe-core`](https://github.com/dequelabs/axe-core/blob/master/doc/developer-guide.md) library to run a series of tests to check for accessibility of content and functionality for DOM nodes in the selector
```

Comment:
only included in this list if the command or helper is used more than a couple of times.